### PR TITLE
Adjust launcher callback ssl

### PR DESCRIPTION
**Pull request by tnederlof**
Closes https://github.com/sol-eng/wbi/issues/142

Added a `DeleteStrings` function in the system package which reads a file, remove lines that are specified and then writes out to a file.

This PR uses `DeleteStrings` to remove the default `launcher-sessions-callback-address=http://localhost:8787` and adds a new `launcher-sessions-callback-address=https://localhost:8787` when enabling SSL.

### CODE REVIEW

**Comment by samcofer** [comment]
This results in an invalid and non-working configuration, why would we want to do this? We can do this correctly, by using the subject/hostname of the server certificate to _maybe_ determine the correct hostname. But if you're changing to SSL, you should probably go ahead and remove port 8787, unless I'm missing something?

**Comment by tnederlof** [comment]
Yea good call on the port. I see what you mean. I struggle to figure out what to do here as leaving it http local host 8787 doesn’t seem right either. I noticed the errors being thrown in verifying the server.

**Comment by samcofer** [comment]
The ideal thing is probably to prompt during the ssl check for the expected
hostname and set it at that point. We can do that after verifying the cert
and before writing the configuration.

**Comment by samcofer** [comment]
On thinking about this further, that might be a better purpose for the mismatched hostname prompt. We can check the match, then ask for the end-user or data scientist-facing URL if they don't match.

**Comment by tnederlof** [comment]
@samcofer In my last commit I added a prompt for the server URL which is then used for the launcher address callback config. I wasn't sure the right value to grab from the SSL certs to try to auto populate this and how reliable that would be so wanted to start here. Although we could move the location of this prompt and/or pass default values from the other SSL functions if you want to try that route.

File: internal/ssl/prompt.go [hunk 1]
```diff
@@ -83,6 +83,22 @@ func PromptSSLFilePath() (string, error) {
 	return target, nil
 }
 
+// PromptServerURL asks users for the server URL
+func PromptServerURL() (string, error) {
+	target := ""
+	messageText := "Server URL that end users will use to access the server (for example, mydomainname.com):"
```
Comment (samcofer) [inline internal/ssl/prompt.go:89]: Can you modify this example to something like: _workbench.mydomainname.com_

Without the hostname it's a little ambiguous if you want to domain name or the hostname.

File: cmd/setup.go [hunk 2]
```diff
@@ -198,7 +198,11 @@ func newSetup(setupOpts setupOpts) error {
 			if err != nil {
 				return fmt.Errorf("issue verifying and configuring SSL: %w", err)
 			}
-			workbench.WriteSSLConfig(certPath, keyPath)
+			serverURL, err := ssl.PromptServerURL()
```
Comment (samcofer) [inline cmd/setup.go:201]: `ssl.PromptServerURL()` is useful in several places in my SSL verification code, can we move this up to run before `ssl.PromptAndVerifySSL()` and I'll refactor my host mismatch code to use it.

**Review by samcofer — COMMENTED** [review]
This looks good, I just had a couple  flow thoughts.

**Comment by tnederlof** [comment]
Good points @samcofer I pushed the changes. I like the idea of merging this first and then you can use the server name in other parts of the code.

File: internal/ssl/prompt.go [hunk 3]
```diff
@@ -86,7 +86,7 @@ func PromptSSLFilePath() (string, error) {
 // PromptServerURL asks users for the server URL
 func PromptServerURL() (string, error) {
 	target := ""
-	messageText := "Server URL that end users will use to access the server (for example, mydomainname.com):"
+	messageText := "Server URL that end users will use to access the server (for example, workbench.mydomainname.com):"
```
Comment (tnederlof) [inline internal/ssl/prompt.go:131]: Add: to access the web interface

Comment (tnederlof) [reply internal/ssl/prompt.go:131]: Add: https:// to the example

**Review by samcofer — APPROVED** [review]
Looks good to me!

**Pull request merged**